stockfish/models.py

```diff
@@ -519,75 +576,167 @@ def get_evaluation(self) -> dict:
+
+            # if we're searching depth and the line is not our desired depth, we're done
+            if (num_nodes == 0) and (int(self._pick(line, "depth")) != self._depth):
+                break
+
+            # if we're searching nodes and the line has less than desired number of nodes, we're done
+            if (num_nodes > 0) and (int(self._pick(line, "nodes")) < self._num_nodes):
+                break
+
+            # gather info on "num_top_moves" multipv lines
+            # multiPV_number = int(current_line[current_line.index("multipv") + 1])
+
+            # should never happen
+            # if multiPV_number > num_top_moves:
+            #     continue
+
+            # should always be the case
+            # if multiPV_number <= num_top_moves:
+
+            # has_centipawn_value = "cp" in current_line
+            # has_mate_value = "mate" in current_line
+
+            # unnecessary check? should never happen, unless major malfunction
+            # if has_centipawn_value == has_mate_value:
+            # raise RuntimeError("Having a centipawn value and mate value should be mutually exclusive.")
+
+            move_evaluation = {
+                # get move
+                "Move": self._pick(line, "pv"),
+                # get cp if available
+                "Centipawn": int(self._pick(line, "cp")) * perspective
+                if "cp" in line
+                else None,
+                # get mate if available
+                "Mate": int(self._pick(line, "mate")) * perspective
+                if "mate" in line
+                else None,
+            }
+
+            # add more info if verbose
+            if verbose:
+                move_evaluation["Time"] = self._pick(line, "time")
+                move_evaluation["Nodes"] = self._pick(line, "nodes")
+                move_evaluation["MultiPVLine"] = self._pick(line, "multipv")
+                move_evaluation["NodesPerSecond"] = self._pick(line, "nps")
+                move_evaluation["SelectiveDepth"] = self._pick(line, "seldepth")
+
+                # add wdl if available
+                if self.does_current_engine_version_have_wdl_option():
+                    move_evaluation["WDL"] = " ".join(
+                        [
+                            self._pick(line, "wdl", 1),
+                            self._pick(line, "wdl", 2),
+                            self._pick(line, "wdl", 3),
+                        ][::perspective]
+                    )
+
+            # add move to list of top moves
+            top_moves.insert(0, move_evaluation)
+
+        # reset MultiPV to global value
+        if old_multipv != self._parameters["MultiPV"]:
+            self._set_option("MultiPV", old_multipv)
+
+        # reset self._num_nodes to global value
+        if old_num_nodes != self._num_nodes:
+            self._num_nodes = old_num_nodes
+
         return top_moves
 
+    def _pick(self, line: list = [], value: str = "", index: int = 1) -> str:
```

Comment:
```suggestion
    def _pick(self, line: list, value: str = "", index: int = 1) -> str:
```
Does it make sense to call the function without this parameter?